### PR TITLE
SPAWN: Expose alpha entity field to server.

### PR DESCRIPTION
**Pull request by dsvensson**
This opens up for static transparent brushes in maps.

### CODE REVIEW

**Comment by tcsabina** [comment]
Hi Nano!

This is again a change that should boil down to ezquake and mvdsv as well (or even more projects? qtv/mvdparser?). The `entvars_t` struct is defined in those projects as well, so if we need to update it in ktx, I would like to see the same change in the other repos as well.

I need to think about how to do this properly. These shared structures could move into a new repository, and that repo will be a submodule in these.

I will get back to you on this...

**Comment by dsvensson** [comment]
Yep, have PRs for all the others on the way. The bummer is that need to have ifdef mvdsv vs ftesv or always padding as ftesv has a ton of other extra stuff in this struct between noise3 and alpha.

**Comment by dsvensson** [comment]
Better approach. Map_Extensions is supported in mvdsv, but not SetExtField. This allows support to be added while keeping backwards compatibility by just removing some ifdef's. Just have to implement SetExtField in mvdsv. Tested with fteqw-sv which unlocks alpha with this change, and mvdsv which prints the unsupported log message.

File: include/g_public.h [hunk 1]
```diff
@@ -200,6 +201,7 @@ typedef enum
 //	F_ENTITY,						// index on disk, pointer in memory
 //	F_ITEM,							// index on disk, pointer in memory
 //	F_CLIENT,						// index on disk, pointer in memory
+	F_EXT_FLOAT,					// float set via SetExtField syscall
```
Comment (dsvensson) [inline include/g_public.h:204]: ah crap, this is used in other places than fields declaration, reshuffling.

File: src/g_syscalls.c [hunk 2]
```diff
@@ -466,15 +466,28 @@ void trap_VisibleTo(intptr_t viewer, intptr_t first, intptr_t len, byte *visible
...
 void trap_SetExtField_f(gedict_t *ed, const char *fieldname, float val)
 {
-	syscall(G_SETEXTFIELD, (intptr_t)ed, (intptr_t)fieldname, PASSFLOAT(val));
+	if (HAVEEXTENSION(G_SETEXTFIELD))
+	{
+		syscall(G_SETEXTFIELD, (intptr_t)ed, (intptr_t)fieldname, PASSFLOAT(val));
+	}
+	else
+	{
+		G_bprint(PRINT_HIGH, "SetExtField(%s, %s, %f) not supported by server\n", ed->classname, fieldname, val);
```
Comment (dsvensson) [inline src/g_syscalls.c:489]: Have to move this logging to some better place, will be spammy with ents that update alpha. Should be printed on map load if alpha ents are in use.

**Pull request merged**